Vungle/VungleRouter.m

```diff
@@ -503,59 +527,71 @@ - (BOOL)validateInfoData:(NSDictionary *)info
     return isValid;
 }
 
-- (void)clearDelegateWithState:(BannerRouterDelegateState)state placementID:(NSString *)placementID
+- (void)clearBannerDelegateWithState:(BannerRouterDelegateState)state
 {
     @synchronized (self) {
-        if (placementID.length > 0) {
-            [self.delegatesDict removeObjectForKey:placementID];
-        } else if (state != BannerRouterDelegateStateUnknown) {
-            NSArray *array = [self.bannerDelegates.keyEnumerator allObjects];
-            for (NSString *key in array) {
-                if ([[self.bannerDelegates objectForKey:key] bannerState] == state) {
-                    [self.bannerDelegates removeObjectForKey:key];
-                }
+        NSArray *array = [self.bannerDelegates.keyEnumerator allObjects];
+        for (NSString *key in array) {
+            if ([[self.bannerDelegates objectForKey:key] bannerState] == state) {
+                [self.bannerDelegates removeObjectForKey:key];
+            }
+        }
+        array = [self.hbBannerDelegates.keyEnumerator allObjects];
+        for (NSString *key in array) {
+            if ([[self.hbBannerDelegates objectForKey:key] bannerState] == state) {
+                [self.hbBannerDelegates removeObjectForKey:key];
             }
         }
     }
 }
 
+- (void)cleanupFullScreenDelegate:(id<VungleRouterDelegate>)delegate
+{
+    @synchronized (self) {
+        NSString *placementID = [delegate getPlacementID];
+        NSString *eventID = [delegate getEventId];
+        if (eventID.length > 0) {
+            [self.hbDelegatesDict removeObjectForKey:eventID];
+        } else if (placementID.length > 0) {
+            [self.delegatesDict removeObjectForKey:placementID];
+        }
+    }
+}
+
+- (void)addToWaitingListWithDelegate:(id<VungleRouterDelegate>)delegate
+{
+    NSString *eventId = [delegate getEventId];
+    if (eventId.length > 0) {
+        [self.hbWaitingListDict setObject:delegate forKey:eventId];
```

Comment:
Do we need to check if the `self.hbWaitingListDict` already has the delegate for the eventId key like below, thanks ^_^

Reply:
Updated.